lib/exhal.ex

```diff
@@ -94,13 +97,15 @@ defmodule ExHal do
   Returns `{:ok,    %ExHal.Document{...}}` if request is an error
           `{:error, %ExHal.Error{...}}` if not
   """
-  def follow_link(a_doc, name, opts \\ %{pick_volunteer: false, tmpl_vars: %{}}) do
-    pick_volunteer? = Dict.get opts, :pick_volunteer, false
-    tmpl_vars = Dict.get opts, :tmpl_vars, %{}
+  def follow_link(a_doc, name, opts \\ %{tmpl_vars: %{}, pick_volunteer: false}, headers \\ []) do
```

Comment:
I'd rather `headers` be an option rather than fourth argument. Also,w ould using Keywords for the options be more idiomatic? Eg:

```elixir
def follow_link(a_doc, name,, opts \\ [tmpl_vars: %{}, pick_volunteer: false, headers: []]
  opts = Map.new(opts)
  ...
```